__tests__/reducers/map.test.js

```
@@ -389,6 +428,104 @@ describe('map reducer', () => {
     });
   });
 
+  it('should handle ADD_FEATURES to add action features to a source feature', () => {
```

You're right, I see how that is probably happening and had to fix another similar bug earlier.  Can you make sure the source is properly defined and let the test fail? Then I'll merge with the known failed test and make a fix.